### PR TITLE
cmd/atlas: support multifile input and directory input schemas

### PR DESCRIPTION
**Pull request by rotemtam**
Multi files/directories in `-f`:

```
atlas schema apply -f 1.hcl -f 2.hcl -u URL
atlas schema apply -f dir/ -u URL
```

Supports directory in project files:
```
-- atlas.hcl --
env "local" {
    url = "URL"
    src = "./schema"
    def_val = "hello"
}
```

todo: multi file support with migrate diff (dir works), e.g `migrate diff --to 1.hcl --to 2.hcl` (API feels weird so didn't add yet)

### CODE REVIEW

File: cmd/atlas/internal/cmdapi/migrate.go [hunk 1]
```diff
@@ -525,14 +521,50 @@ func to(ctx context.Context, client *sqlclient.Client) (migrate.StateReader, err
...
+				return nil, err
+			}
+		}
 	}
 	return p, nil
 }
 
+// mayParse will parse the file in path if it is an HCL file not named "atlas.hcl".
+func mayParse(p *hclparse.Parser, path string) error {
+	if n := filepath.Base(path); n == projectFileName || !strings.HasSuffix(n, ".hcl") {
```
Comment (masseelch) [inline cmd/atlas/internal/cmdapi/migrate.go:559]:
```suggestion
	if n := filepath.Base(path); n == projectFileName || filepath.Ext(n) != ".hcl" {
```

File: cmd/atlas/internal/cmdapi/migrate.go [hunk 2]
```diff
@@ -525,14 +521,50 @@ func to(ctx context.Context, client *sqlclient.Client) (migrate.StateReader, err
...
+				if f.IsDir() {
+					continue
+				}
+				if err := mayParse(p, filepath.Join(path, f.Name())); err != nil {
+					return nil, err
+				}
+			}
+		default:
+			if err := mayParse(p, path); err != nil {
+
```
Comment (a8m) [inline cmd/atlas/internal/cmdapi/migrate.go:549]:
```suggestion
```

File: cmd/atlas/internal/cmdapi/migrate.go (hunk 1 above)
Comment (masseelch) [reply cmd/atlas/internal/cmdapi/migrate.go:559]: I prefer filepath operations on filepaths

File: cmd/atlas/internal/cmdapi/migrate.go [hunk 3]
```diff
@@ -525,14 +521,50 @@ func to(ctx context.Context, client *sqlclient.Client) (migrate.StateReader, err
...
 
-func parseHCLBytes(b []byte) (*hclparse.Parser, error) {
+// parseHCL paths parses the HCL files in the given paths. If a path represents a directory,
+// its direct descendants will be considered, skipping any subdirectories. Files named "atlas.hcl"
+// are skipped as well/
+func parseHCLPaths(paths []string) (*hclparse.Parser, error) {
 	p := hclparse.NewParser()
-	if _, diag := p.ParseHCL(b, ""); diag.HasErrors() {
-		return nil, diag
+	for _, path := range paths {
```
Comment (a8m) [inline cmd/atlas/internal/cmdapi/migrate.go:529]: Why not `filepath.Walk` with `filepath.SkipDir` to skip nested directories?

File: cmd/atlas/internal/cmdapi/migrate.go [hunk 4]
```diff
@@ -525,14 +521,50 @@ func to(ctx context.Context, client *sqlclient.Client) (migrate.StateReader, err
...
+// mayParse will parse the file in path if it is an HCL file not named "atlas.hcl".
+func mayParse(p *hclparse.Parser, path string) error {
+	if n := filepath.Base(path); n == projectFileName || !strings.HasSuffix(n, ".hcl") {
+		return nil
+	}
+	if _, diag := p.ParseHCLFile(path); diag.HasErrors() {
+		return diag
+	}
+	return nil
+}
```
Comment (a8m) [inline cmd/atlas/internal/cmdapi/migrate.go:566]: 💯

**Review by masseelch — APPROVED** [review]

File: cmd/atlas/internal/cmdapi/schema.go [hunk 5]
```diff
@@ -136,7 +136,7 @@ func init() {
 	// Schema apply flags.
 	schemaCmd.AddCommand(SchemaApply)
 	SchemaApply.Flags().SortFlags = false
-	SchemaApply.Flags().StringVarP(&ApplyFlags.File, fileFlag, "f", "", "[/path/to/file] file containing the HCL schema.")
+	SchemaApply.Flags().StringSliceVarP(&ApplyFlags.Paths, fileFlag, "f", nil, "[/path/to/schemas] file or directory containing the HCL files")
```
Comment (a8m) [inline cmd/atlas/internal/cmdapi/schema.go:139]:
```suggestion
	SchemaApply.Flags().StringSliceVarP(&ApplyFlags.Paths, fileFlag, "f", nil, "[paths...] file or directory containing the HCL files")
```

**Review by a8m — APPROVED** [review]
💯

File: cmd/atlas/internal/cmdapi/migrate.go (hunk 1 above)
Comment (a8m) [reply cmd/atlas/internal/cmdapi/migrate.go:559]: 👍

**Review by a8m — APPROVED** [review]

**Pull request merged**